### PR TITLE
python-package: refine plot_tree

### PR DESCRIPTION
**Pull request by henry0312**
This change splits plot_tree to two methods:
1. method of creating digraph of tree
2. method of plotting the digraph with matplotlib

Some users want (1) when a tree has many leaves.

### CODE REVIEW

**Comment by wxchan** [comment]
can you paste some plotted images to show the improvements here?

File: docs/Python-API.md [hunk 1]
```diff
@@ -1045,3 +1045,27 @@ The methods of each Class is in alphabetical order.
...
+    ----------
+    booster : Booster, LGBMModel
+        Booster or LGBMModel instance.
+    tree_index : int, default 0
+        Specify tree index of target tree.
+    graph_attr : dict
+        Mapping of (attribute, value) pairs for the graph.
+    node_attr : dict
+        Mapping of (attribute, value) pairs set for all nodes.
+    edgeattr : dict
```
Comment (wxchan) [inline docs/Python-API.md]: '_' missing

**Comment by henry0312** [comment]
@wxchan 

Current (with matplotlib)
image

With (1), we can get pdf like this (obviously better than png)
[tree2.pdf](https://github.com/Microsoft/LightGBM/files/907486/tree2.pdf)

**Comment by henry0312** [comment]
@wxchan @guolinke ready for review (and merge?)

**Comment by wxchan** [comment]
I think it's ok to merge.

**Comment by henry0312** [comment]
@guolinke please review

**Comment by henry0312** [comment]
thank you.

**Pull request merged**